cpp/oneapi/dal/backend/primitives/objective_function/test/spmd_fixture.hpp

```diff
@@ -0,0 +1,226 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "fixture.hpp"
+#include "oneapi/dal/test/engine/tables.hpp"
+#include "oneapi/dal/test/engine/io.hpp"
+
+namespace oneapi::dal::backend::primitives::test {
+
+namespace te = dal::test::engine;
+namespace de = dal::detail;
+
+template <typename Param>
+class logloss_spmd_test : public logloss_test<Param> {
+public:
+    using float_t = Param;
+    using comm_t = te::thread_communicator<spmd::device_memory_access::usm>;
+
+    std::vector<std::pair<table, ndview<std::int32_t, 1>>>
+    split_input_data(const table& data, ndview<std::int32_t, 1> labels, std::int64_t split_count) {
+        ONEDAL_ASSERT(split_count > 0);
+        const std::int64_t row_count = data.get_row_count();
+        const std::int64_t column_count = data.get_column_count();
+        const std::int64_t block_size_regular = row_count / split_count;
+        const std::int64_t block_size_tail = row_count % split_count;
+
+        std::vector<std::pair<table, ndview<std::int32_t, 1>>> result(split_count);
+
+        std::int64_t row_offset = 0;
+        for (std::int64_t i = 0; i < split_count; i++) {
+            const std::int64_t tail = std::int64_t(i + 1 == split_count) * block_size_tail;
+            const std::int64_t block_size = block_size_regular + tail;
+            // std::cout << i << ": " << row_offset << " " << block_size << std::endl;
+            REQUIRE(block_size > 0);
```

Comment:
Please use natural order wherever it is possible:
```suggestion
            REQUIRE(0l < block_size);
```